common/set.h

```diff
@@ -0,0 +1,334 @@
+  };
+
+  // Implicitly convertible to the relevant view type.
+  //
+  // NOLINTNEXTLINE(google-explicit-constructor): Designed to implicitly decay.
+  operator ViewT() const { return this->view_impl(); }
+
+  // We can't chain the above conversion with the conversions on `ViewT` to add
+  // const, so explicitly support adding const to produce a view here.
+  //
+  // NOLINTNEXTLINE(google-explicit-constructor): Designed to implicitly decay.
+  operator SetView<const KeyT, KeyContextT>() const { return ViewT(*this); }
+
+  // Convenience forwarder to the view type.
+  template <typename LookupKeyT>
+  auto Contains(LookupKeyT lookup_key,
+                KeyContextT key_context = KeyContextT()) const -> bool {
+    return ViewT(*this).Contains(lookup_key, key_context);
+  }
+
+  // Convenience forwarder to the view type.
+  template <typename LookupKeyT>
+  auto Lookup(LookupKeyT lookup_key,
+              KeyContextT key_context = KeyContextT()) const -> LookupResult {
+    return ViewT(*this).Lookup(lookup_key, key_context);
+  }
+
+  // Convenience forwarder to the view type.
+  template <typename CallbackT>
+  void ForEach(CallbackT callback) {
+    return ViewT(*this).ForEach(callback);
+  }
+
+  // Convenience forwarder to the view type.
+  auto CountProbedKeys(KeyContextT key_context = KeyContextT()) const
+      -> ssize_t {
+    return ViewT(*this).CountProbedKeys(key_context);
+  }
+
+  // Insert a key into the set. If the key is already present, no insertion is
+  // performed and that present key is available in the result. Otherwise a new
+  // key is inserted and constructed from the argument and available in the
+  // result.
+  template <typename LookupKeyT>
+  auto Insert(LookupKeyT lookup_key, KeyContextT key_context = KeyContextT())
+      -> InsertResult;
+
+  // Insert a key into the set and call the provided callback to allow in-place
+  // construction of the key if not already present. The lookup key is passed
+  // through to the callback so it needn't be captured and can be kept in a
+  // register argument throughout.
+  //
+  // Example:
+  // ```cpp
+  //   m.Insert("widget", [](MyStringViewType lookup_key, void* key_storage) {
+  //     new (key_storage) MyStringType(lookup_key);
+  //   });
+  // ```
+  template <typename LookupKeyT, typename InsertCallbackT>
+  auto Insert(LookupKeyT lookup_key, InsertCallbackT insert_cb,
+              KeyContextT key_context = KeyContextT()) -> InsertResult
+    requires std::invocable<InsertCallbackT, LookupKeyT, void*>;
+
+  // Erase a key from the set.
+  template <typename LookupKeyT>
+  auto Erase(LookupKeyT lookup_key, KeyContextT key_context = KeyContextT())
+      -> bool;
+
+  // Clear all key/value pairs from the set but leave the underlying hashtable
+  // allocated and in place.
+  void Clear();
+
+ protected:
+  using ImplT::ImplT;
+};
+
+// A data structure for a set of keys.
+//
+// This set also supports small size optimization (or "SSO"). The provided
```

Comment:
Not clear what "also" is referring to in this context.

```suggestion
// This set supports small size optimization (or "SSO"). The provided
```